crates/codegen/parser_runtime/src/templates/kinds.tera

```diff
@@ -0,0 +1,63 @@
+#[cfg(feature = "slang_napi_interfaces")]
+use { napi::bindgen_prelude::*, napi_derive::napi };
+
+#[derive(
+    Debug,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    strum_macros::AsRefStr,
+    strum_macros::Display,
+    strum_macros::EnumString,
+)]
+#[cfg_attr( feature = "slang_napi_interfaces", /* derives `Clone` and `Copy` */ napi(string_enum, namespace = "kinds") )]
+#[cfg_attr( not(feature = "slang_napi_interfaces"), derive(Clone, Copy))]
+pub enum ProductionKind {
+    {%- for variant in code.production_kinds -%}
+        {# variant.documentation | indent(prefix = "/// ", first = true, blank = true) #}
+        {{ variant }},
+    {%- endfor -%}
+}
+
+#[derive(
```

Comment:
I suggest keeping the template to one `enum` type, and generating one file for each "kind". I did that separation originally in #543 because it makes reviewing changes much easier, comparing additions/removals across the three types.

Reply:
I would like to eliminate the template with quote! and the text transform (`to_pretty_string()` as you suggested).

Prefer to push this for now.

Reply:
Sure. Let's spike that later, but for the near future, I suggest keeping them split if not too much work. It is hard to keep track of changes if they are all in one file.

Reply:
I agree, but at this point, later PR.

Reply:
#573